### PR TITLE
feat(ha): csi node should label nodes with nvme ana capability, node agent should not report failed paths if ana is not supported

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
- When CSI nodes start up it will detect the presence of `nvme_tcp` kernel modules and exit with a message.
- When CSI nodes startup it will label the k8s node with the nvme-ana capability.
- If nvme-ana is not supported HA nodes will not report failed paths, it will just log.

### CODE REVIEW

File: control-plane/agents/src/bin/ha/node/detector.rs [hunk 1]
```diff
@@ -136,6 +136,15 @@ impl PathRecord {
     }
 
     fn report_live(&mut self) {
+        if self.state != PathState::Good {
+            tracing::info!(
+                from=%self.state,
+                to=%PathState::Good,
+                target=self.nqn,
+                path=self.path,
+                "Target state transition",
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/ha/node/detector.rs]: Something like this

```suggestion
                "Target path has been fixed",
```

File: control-plane/agents/src/bin/ha/node/reporter.rs [hunk 2]
```diff
@@ -112,7 +116,11 @@ impl PathReporter {
 
     /// Reports the given NVMe NQN as a failed path.
     pub fn report_failed_path(&self, nqn: String) {
-        self.channel.send(nqn).ok();
+        if !self.ana_enabled {
+            info!("nvme ana multi-pathing is not supported, not reporting the failed path: {nqn}")
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/ha/node/reporter.rs]:
```suggestion
            tracing::warn!(path.nqn, "NVMe ANA multipathing is not supported, not reporting the failed path");
```

File: control-plane/csi-driver/src/bin/node/main_.rs [hunk 3]
```diff
@@ -167,6 +175,20 @@ pub(super) async fn main() -> anyhow::Result<()> {
...
+    if let Err(error) = crate::dev::nvmf::check_nvme_tcp_module() {
+        anyhow::bail!("Failed to detect nvme_tcp kernel module. Run `modprobe nvme_tcp` to load the kernel module. {}", error);
+    }
+
+    if platform::current_plaform_type() == platform::PlatformType::K8s {
+        check_ana_and_label_node(
+            matches
+                .get_one::<String>("node-name")
+                .expect("required")
+                .as_str(),
```
Comment (tiagolobocastro) [inline control-plane/csi-driver/src/bin/node/main_.rs]:
```suggestion
```
Does it not coerce into &str?

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/csi-driver/src/bin/node/main_.rs (hunk 3 above)
Comment (Abhinandan-Purkait) [reply control-plane/csi-driver/src/bin/node/main_.rs]: Yes, it does. Done.

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**